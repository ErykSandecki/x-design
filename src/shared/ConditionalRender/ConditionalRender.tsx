import { Activity, FC, JSX } from 'react';

export type TConditionalRenderProps = {
  component: () => JSX.Element;
  visible: boolean;
};

export const ConditionalRender: FC<TConditionalRenderProps> = ({ component: Component, visible }) => (
  <Activity mode={visible ? 'visible' : 'hidden'}>
    <Component />
  </Activity>
);

export default ConditionalRender;
