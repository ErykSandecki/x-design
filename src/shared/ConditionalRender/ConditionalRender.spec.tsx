import { render } from '@testing-library/react';

// components
import ConditionalRender from './ConditionalRender';

describe('ConditionalRender snapshots', () => {
  it('should render ConditionalRender', () => {
    // before
    const { asFragment } = render(<ConditionalRender component={<>Component</>} visible />);

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render not visible', () => {
    // before
    const { asFragment } = render(<ConditionalRender component={<>Component</>} visible={false} />);

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
