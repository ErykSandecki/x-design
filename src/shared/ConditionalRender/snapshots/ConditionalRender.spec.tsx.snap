// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionalRender snapshots should render ConditionalRender 1`] = `
<DocumentFragment>
  Component
</DocumentFragment>
`;

exports[`ConditionalRender snapshots should render not visible 1`] = `<DocumentFragment />`;
